### DISCUSSION TITLE
How array slice is processed

### DISCUSSION THREAD

**Discussion by jamhot1** [kernc/backtesting.py#220] [Q&A]
I wanted to evaluate a range of values created in a custom df column during Next() 

when using the below example, am I looking at the entire array (very last Five values) or the array from the point reached at Next()

a[-5:]   # last Five items

I seem to be getting unexpected behavior. My assumption is last Five from current Next() line?

**Comment by kernc** [comment]
> the array from the point reached at Next()

The array _from beginning_ and _to_ the point reached at `next()`.

Hopefully, you can make use of these:
```py
>>> from backtesting import Backtest, Strategy
... from backtesting.test import GOOG
...
... class S(Strategy):
...     def init(self): pass
...     def next(self): print(self.data.Close.size)
...
... Backtest(GOOG.iloc[:5], S).run()
2
3
4
5

...
```
https://github.com/kernc/backtesting.py/blob/1ee5670d03b26680d9add5c96a9f98e44fc722a2/backtesting/backtesting.py#L1147-L1161

This, of course, only works for [`Strategy.data` arrays](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.data) and indicator arrays computed via [`Strategy.I()`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.I).

**Reply by jamhot1** [reply]
Thanks for clearing that up :) After some playing around, I think it was because I was getting negative values in the last 5 tested. I need to investigate more, or simply replace for a positive values. Right now I am unsure what the negative values are doing to create the unexpected behaviour.
